### ISSUE TITLE
[bench request]: Mirror (DeBERTa-v3-large, 436M) with classification probabilities and reproducible API

### ISSUE THREAD

**Issue by Spico197** [fstandhartinger/jevbench#13]
Request to benchmark **Mirror (DeBERTa-v3-large, 436M), with a single-choice probability decoder**, on the complete frozen JevBench v1.2 set.

**Runnable public code:** https://github.com/Spico197/mirror-jevbench — submission revision `93c49aa`.

**Setup and serving:** https://github.com/Spico197/mirror-jevbench/blob/93c49aa/MIRROR-SERVING.md

**Results and evidence:** https://github.com/Spico197/mirror-jevbench/blob/93c49aa/MIRROR-API-RESULTS.md — per-item distributions, raw HTTP responses, hashes, runtime manifests and verification script are included under `results/api-v3/`.

### Model and interface

This uses the original public `Mirror_Pretrain_AllExcluded_2` weights from [Spico/Mirror](https://huggingface.co/spaces/Spico/Mirror), unchanged. The wrapper exposes `GET /health` and `POST /v1/systemone` for choice, noul and score, so your existing `typesafe` adapter works with `--key-env ''`. It can run on your own GPU; a submitter-operated endpoint is not needed for held-out questions. No benchmark request bodies are logged or sent elsewhere by the server.

```bash
git clone https://github.com/Spico197/mirror-jevbench
cd mirror-jevbench
git checkout 93c49aa
python3.10 -m venv .venv
source .venv/bin/activate
pip install -r requirements-mirror.txt
python scripts/setup_mirror_submission.py
python mirror_server.py --host 127.0.0.1 --port 8000 --device cuda
curl http://127.0.0.1:8000/health
```

Setup downloads pinned public source, tokenizer and checkpoint, and verifies the checkpoint SHA256 (`b81f6f1eed9792bf7767c1c9bd14ade382c64f120266b9eb3dc03cbafa952703`). We tested on H200 in FP32, batch size 1, TF32 off, PyTorch 2.6.0+cu124 and transformers 4.28.0. The model has 435,757,059 parameters; this is an encoder model, not a generative LLM. CPU loading is supported, but we have not measured CPU performance.

### What changed and what the probabilities mean

Mirror originally uses thresholded span decoding, which returned no classification label for 99/231 public decisions. For each candidate marker position `i`, the wrapper now computes:

```text
s_i = logsigmoid(NNW[i,i]) + logsigmoid(THW[i,i])
p_i = exp(s_i - max(s)) / sum_j exp(s_j - max(s))
```

These are the two edges required for a singleton classification span; NSW connects different span parts. Temperature is fixed at 1, with no fitted calibration. The returned choice is always the probability argmax, with lexical tie-breaking. Noul returns P(yes); ordinal questions return the full indexed distribution and expected level.

We describe these explicitly as **derived normalized classification-span scores**, not an originally trained categorical head or a claim of calibration. They are not generated/verbalized probabilities or token-generation logprobs. Your stock TypeSafe adapter labels endpoint probabilities `native`; the API and manifests carry the more precise method. Please use the provenance category appropriate to your rules.

### Public results through the unchanged TypeSafe adapter

We ran all 231 public items via actual serial HTTP using your harness revision `7ce310c7262ed49cc85853339a8a42459298e3f3` and dataset hash `dc3995d8ae1e2fc8e81ce38431add509eb8bb39b85aadfd0c7c32079382dde51`.

| Configuration | Easy | Standard/public original | Hard/public | Overall |
|---|---:|---:|---:|---:|
| **Strict 512-token context (requested submission)** | 42/48 (87.5%) | 33/72 (45.8%) | 20/111 (18.0%) | **95/231 (41.1%)** |
| Upstream truncation, diagnostic only | 42/48 (87.5%) | 33/72 (45.8%) | 41/111 (36.9%) | 116/231 (50.2%) |

The default strict configuration returns HTTP 422 for 59 inputs over 512 tokens, counting them as misses. The full length includes instruction, complete rubric, candidate labels, state and special tokens. The diagnostic mode (`--context-policy truncate`) retains the upstream state-prefix truncation behavior. Both policies were specified before this HTTP run; we are requesting the strict configuration, not selecting the higher-scoring mode.

Strict ECE is 0.2893 and multiclass Brier is 0.7256 on the 172 returned distributions; hard ECE is 0.4530 on 52 returned distributions. Rejections stay in the accuracy denominator but have no distribution for calibration. Truncation ECE is 0.2834 overall and 0.3508 on all 111 public hard items. There are no public distribution-gold items, so we have no distribution-fidelity result or official four-axis score.

The original direct-run repaired result was 117/231 (50.6%). That run supplied `task.labels` order directly. The standard TypeSafe request contains only the criteria object, so the API uses its insertion order for choice labels. We reran everything through the actual adapter and report the changed results above; we do not claim those earlier predictions as the API result.

### Timing, cost and disclosure

- Serial loopback HTTP after model load and warmup: strict p50/p95 0.0160/0.0241 s (includes fast context rejections); truncation 0.0169/0.0378 s. These are local verification measurements, not latency from your server or under production load.
- No hosted tariff exists for this wrapper; please apply your documented encoder cost-estimation policy. It does not generate output tokens, and `usage.input_tokens` reports actual encoded tokens for successful requests.
- The fix was developed after observing missing predictions on the public set. Probability normalization was proposed after seeing the repaired public accuracy. No model fine-tuning, prompt search or temperature fitting was performed on JevBench; we do not assert that original pretraining has no overlap.
- Wrapper code is Apache-2.0; upstream source and weights retain their respective licenses/terms.
- 28 focused tests passed, including choice/noul/score round trips through your HTTP adapter. All 462 rows across both runs were independently checked against raw-response hashes, recomputed scores and context handling.

Could you evaluate this pinned strict configuration on the full 534 decisions and include it with the probability-conversion and context-limit disclosures? The public code and weights can run entirely on your hardware.

**Comment by fstandhartinger** [comment]
Thanks — this is logged for reachability checks and the JevBench measurement queue. We’ll only make benchmark claims after our own run.

- Harold

**Comment by fstandhartinger** [comment]
Congratulations on the JevBench v1.4 release, Bluusun! Thanks for building and sharing your system.

- Mirror: #66 at 2.11

Mirror is newly public in v1.4. V1.4 blends in 308 fresh sealed decisions, uses an equal-weight harmonic mean, and applies separate Speed and Cost gates. For your leading row, sealed accuracy was 9.1% versus 41.1% on the public slice. The >25-point gap adjustment applied; the below-50 Intelligence gate also applied.

Full board and method: https://benchmarkheaven.com/jev-models
